src/main/java/gregtech/common/metatileentities/miner/MetaTileEntityLargeMiner.java

```diff
@@ -0,0 +1,659 @@
+        ModularUI.Builder b = super.createUITemplate(entityPlayer);
+        b.widget(new AdvancedTextWidget(9, 20, l -> addRealDisplayText(l, stateHolder.get()), 0xFFFFFF)
+                .setMaxWidthLimit(181)
+                .setClickHandler(this::handleDisplayClick));
+        this.uiConfigModeStateHolder = null;
+        return b;
+    }
+
+    @Nonnull
+    @Override
+    protected Widget getFlexButton(int x, int y, int width, int height) {
+        AtomicBoolean stateHolder = this.uiConfigModeStateHolder;
+        if (stateHolder == null) return super.getFlexButton(x, y, width, height);
+        return new ImageCycleButtonWidget(x, y, width, height, GuiTextures.BUTTON_MINER_CONFIG_MODE,
+                stateHolder::get, stateHolder::set)
+                .setTooltipHoverString(i -> i == 0 ?
+                        "gregtech.machine.miner.button.tooltip.info" :
+                        "gregtech.machine.miner.button.tooltip.config");
+    }
+
+    @Override
+    protected void addDisplayText(List<ITextComponent> textList) {} // unused
+
+    protected void addRealDisplayText(List<ITextComponent> textList, boolean configMode) {
+        if (!configMode) {
+            super.addDisplayText(textList);
+        }
+
+        if (!this.isStructureFormed()) return;
+
+        if (!configMode) {
+            if (energyContainer != null && energyContainer.getEnergyCapacity() > 0) {
+                int energyContainer = getEnergyTier();
+                long maxVoltage = GTValues.V[energyContainer];
+                String voltageName = GTValues.VNF[energyContainer];
+                textList.add(new TextComponentTranslation("gregtech.multiblock.max_energy_per_tick", maxVoltage, voltageName));
+            }
+        }
+
+        ITextComponent areaText;
+        int area, maxArea;
+        if (this.minerLogic.isChunkMode()) {
+            area = this.minerLogic.getCurrentChunkDiameter();
+            maxArea = this.minerLogic.getMaximumChunkDiameter();
+            areaText = new TextComponentTranslation("gregtech.machine.miner.display.working_area.chunks", area, area);
+        } else {
+            area = this.minerLogic.getCurrentDiameter();
+            maxArea = this.minerLogic.getMaximumDiameter();
+            areaText = new TextComponentTranslation("gregtech.machine.miner.display.working_area", area, area);
+        }
+
+        if (configMode) {
+            areaText.appendText(" ").appendSibling(incrButton(area, maxArea, MinerUtil.DISPLAY_CLICK_AREA_INCR))
+                    .appendText(" ").appendSibling(decrButton(area, 1, MinerUtil.DISPLAY_CLICK_AREA_DECR))
+                    .appendText(" ").appendSibling(previewAreaButton(this.minerLogic.isPreviewEnabled()));
+        }
+
+        textList.add(areaText);
+
+        if (configMode) {
+            int yLimit = this.minerLogic.getYLimit();
+            ITextComponent value;
+            ITextComponent hoverText;
+            if (yLimit > 0) {
```

Comment:
This needs to be limited by the height of the multiblock above bedrock, or something that just prevents it from increasing forever